WIP: Allow user to add mentions and a description body to the scry

hardfire opened the pull request:
This allows the user to add a description and mentions to the scry. 
The description is not shown anywhere (AFAIK).
I'll test the mention/notification in a while.

hardfire commented:
![image](https://user-images.githubusercontent.com/513457/47279725-a7463c00-d5f2-11e8-9eed-9b56896590b0.png)
woot! it works!!

mixmix commented on views/new-steps/add-mentions.js:
```diff
@@ -0,0 +1,33 @@
+const { h, computed } = require('mutant')
+const Recipients = require('../component/recipients')
+
+module.exports = function AddMentions ({ state, prev, next, myKey, suggest, avatar }) {
```
This is more than just add-mentions, so perhaps it could be named something different.

I need to spin this branch up and have a look at the design to see how this all fits in. It could be that things content could live as part of an existing step ok. Will come back and write more

hardfire replied:
this remains as `add-mentions` now

mixmix commented on views/component/recipients.js:
```diff
@@ -0,0 +1,82 @@
+    }
+  } = opts
+
+  return h('GatheringRecipients', [
```
we can't leave this as GatheringRecipients, otherwise it will collide with the styles of Gatherings ... 
Make this ScryRecipients

hardfire replied:
oh yes! changed.

mixmix commented on views/new.js:
```diff
@@ -28,14 +34,22 @@ module.exports = function ScryNew (opts) {
 
   function publish () {
-    const { title, days, times, closesAt } = resolve(state)
+    const { title, days, times, closesAt, body, mentions } = resolve(state)
```
I think I would use `description` in the state because it's easier to know what that means, then in the opts say `body: description` because that's what's scuttle-poll wants.
This is a minor point though

hardfire replied:
renamed back to description, definitely much readable for the rest of the code.

mixmix commented on views/new-steps/add-mentions.js:
```diff
@@ -0,0 +1,33 @@
+      Recipients({ state, myKey, suggest, avatar })
+    ]),
+    h('div.actions', [
+      prev ? h('button', { 'ev-click': prev }, 'Cancel') : null,
```
change `Cancel` to `Back`

hardfire replied:
nice catch! changed.

mixmix commented on views/component/recipients.js:
```diff
@@ -0,0 +1,82 @@
+    suggest = {},
+    avatar = (id, size) => id,
+    i18n = (key) => {
+      const translations = { 'gathering.action.addMoreRecps': '(optional) invite to this scry' }
```
change this from gatherings.action.addMoreRecps to something scry related

hardfire replied:
renamed to `scry.action.invite`

mixmix commented:
This is looking really good. The functionality works (though I didn't try publishing anything yet, I trust you from your screenshot).

There are a few minor changes above that I'd like made.

In terms of design I had 2 changes I'd like made : 
1. put the "description" on the invoke step, so that you enter the title, the description (optional) and when it closes. To me this is the "setting up" stage when the person is thinking about what the event is about, and title and description go together well. (closesAt probably needs a better home, but that's another problem for another time!)
2. The mentions need some little changes : 
  - ![image](https://user-images.githubusercontent.com/2665886/47423613-f900e900-d7e1-11e8-8b46-92e39282b17a.png)

hardfire commented:
Thanks for the detailed review. 
1. moved `description` to the first screen. I'm still not sure when one would be ready to write a description, after choosing the dates and invitees or before. Good thing is they can go back and change the content so it should be okay with the title.
2. Changed the design. Definitely more clear now.

mixmix commented:
excellent!
merged 

# :pineapple: :tropical_fish:
